### PR TITLE
linux: search for system libc++

### LINKED ISSUE #16754: linux: link system llvm/clang/lld may cause two distinct libc++ linkages
- when linking system llvm/clang/lld
- on linux distros with `libc++` as their system choice for c++ (such as chimera linux)
- zig only searches for system `libstdc++` and fallback is to use bundled `libc++`

`libc++.so` is never found and zig .cpp artifacts link to zig-bundled `libc++.a`, while llvm/clang/lld implicitly link to system `libc++.so`. This results in 2 distinct copies of (some) globals in the zig exe artifact.

chimera-linux report: https://github.com/chimera-linux/cports/pull/263

### PR DESCRIPTION
**Pull request by mikdusan**
When linking system llvm/clang/lld, add system libc++ to the searchlist. This is needed for certain distros (eg. chimera linux).

closes #16754

- tested on chimera-linux
- tested on archlinux

### CODE REVIEW

File: build.zig [hunk 1]
```diff
@@ -630,14 +630,18 @@ fn addCmakeCfgOptionsToExe(
...
-                // back to -lc++ and cross our fingers.
-                addCxxKnownPath(b, cfg, exe, b.fmt("libstdc++.{s}", .{lib_suffix}), "", need_cpp_includes) catch |err| switch (err) {
-                    error.RequiredLibraryNotFound => {
-                        exe.linkLibCpp();
-                    },
-                    else => |e| return e,
-                };
+                // First we try to link against system libstdc++ or libc++.
+                // If that doesn't work, we fall to -lc++ and cross our fingers.
+                var found = false;
```
Comment (andrewrk) [inline build.zig:635]: `const found = for (...) ...`

**Review by andrewrk — CHANGES_REQUESTED** [review]

File: build.zig (hunk 1 above)
Comment (mikdusan) [reply build.zig]: Done. (me always with the mutables).

File: build.zig [hunk 2]
```diff
@@ -630,14 +630,16 @@ fn addCmakeCfgOptionsToExe(
...
-                // First we try to link against gcc libstdc++. If that doesn't work, we fall
-                // back to -lc++ and cross our fingers.
-                addCxxKnownPath(b, cfg, exe, b.fmt("libstdc++.{s}", .{lib_suffix}), "", need_cpp_includes) catch |err| switch (err) {
-                    error.RequiredLibraryNotFound => {
-                        exe.linkLibCpp();
-                    },
-                    else => |e| return e,
-                };
+                // First we try to link against system libstdc++ or libc++.
+                // If that doesn't work, we fall to -lc++ and cross our fingers.
```
Comment (rohlem) [inline build.zig:634]: typo I think? "fall" -> "fall back"

**Comment by andrewrk** [comment]
Thanks! Great work :+1:

**Pull request merged**